### PR TITLE
Add test of entire cloze pipeline outside db/genanki

### PR DESCRIPTION
**Pull request by wclausen**
Adds testing seam in form of new method `_generate_clozed_highlight_notes`. Exercises the new method in tests that can be reliably run on every merge to validate our cloze pipeline continues to function as expected, at high level

In the process of adding the test, it because clear that using
`list("some string")`
is not the same as
`["some string"]`
and we definitely want the latter

### CODE REVIEW

File: tests/test_highlight_clozer.py [hunk 1]
```diff
@@ -4,6 +4,10 @@
 # GIVEN keyword exists with punctuation in sentence
 # WHEN getting the cloze version of the sentence
 # THEN returns cloze that clozes the keyword and retains un-clozed punctuation
```
Comment (z1lc) [inline tests/test_highlight_clozer.py:6]: we probably want this below the imports

**Review by z1lc — APPROVED** [review]

**Comment by z1lc** [comment]
What's the difference between `list("some string")` vs `["some string"]`? From looking online, it seems like it's a [performance difference](https://stackoverflow.com/questions/30216000/why-is-faster-than-list)? Surprised if there's a functional difference.

**Comment by wclausen** [comment]
> What's the difference between `list("some string")` vs `["some string"]`? From looking online, it seems like it's a [performance difference](https://stackoverflow.com/questions/30216000/why-is-faster-than-list)? Surprised if there's a functional difference.

maybe we are doing something weird with `list()` but `list("example")` produces `['e', 'x', 'a', 'm', 'p', 'l', 'e']` instead of `["example"]`. Perhaps this is because a string is iterable?

File: tests/test_highlight_clozer.py (hunk 1 above)
Comment (wclausen) [reply tests/test_highlight_clozer.py:6]: 👍

**Pull request merged**